packages/astro/src/core/build/generate.ts

```diff
@@ -70,17 +75,37 @@ export function chunkIsPage(
 
 export async function generatePages(opts: StaticBuildOptions, internals: BuildInternals) {
 	const timer = performance.now();
-	info(opts.logging, null, `\n${bgGreen(black(' generating static routes '))}`);
-
 	const ssr = opts.settings.config.output === 'server';
 	const serverEntry = opts.buildConfig.serverEntry;
 	const outFolder = ssr ? opts.buildConfig.server : getOutDirWithinCwd(opts.settings.config.outDir);
+
+	if (opts.settings.config.experimental.prerender) {
+		// HACK: force output to be treated as ESM
```

Comment:
why is this needed?

Reply:
Test failures when the fixture did not use `"type": "module"`, but I should confirm that this is still needed.

Reply:
I think the way we make sure it is run as ESM is by adding the .mjs extension. Has that changed?

Reply:
Nope, I think this was a weird edge case as I was moving things around. Going to remove and check CI.